### PR TITLE
Adds forward looking example, handle parse error, update tests

### PR DESCRIPTION
**Pull request by collindutter**
https://github.com/griptape-ai/griptape/issues/239

### CODE REVIEW

**Comment by collindutter** [comment]
I tried adding an activity for getting a time delta, but the LLM got confused on when to use `get_datetime_delta` and when to use `get_relative_datetime`. I think this should be a future PR.
```python
    @activity(config={
        "uses_default_memory": False,
        "description": "Can be used to get get a time delta between two dates.",
        "schema": Schema({
            Literal(
                "date_1",
                description='The first date. Must be a valid date string.'
            ): str,
            Literal(
                "date_2",
                description='The second date. Must be a valid date string.'
            ): str
        })
    })
    def get_datetime_delta(self, params: dict) -> BaseArtifact:
        try:
            date_1 = parse(params["values"]["date_1"])
            date_2 = parse(params["values"]["date_2"])
            if not date_1:
                return ErrorArtifact("invalid date string for date_1")
            if not date_2:
                return ErrorArtifact("invalid date string for date_2")

            delta = date_1 - date_2

            return TextArtifact(delta)
        except Exception as e:
            return ErrorArtifact(f"error getting current datetime: {e}")
```

File: griptape/tools/date_time/tool.py [hunk 1]
```diff
@@ -25,14 +25,17 @@ def get_current_datetime(self, _: dict) -> BaseArtifact:
...
         })
     })
     def get_relative_datetime(self, params: dict) -> BaseArtifact:
         try:
             date_string = params["values"]["relative_date_string"]
             relative_datetime = parse(date_string)
+            if not relative_datetime:
+                return ErrorArtifact("invalid date string")
 
             return TextArtifact(str(relative_datetime))
```
Comment (vasinov) [inline griptape/tools/date_time/tool.py:40]: Let's always have an `if-else` structure for all returns. General rule should be "no early returns" as they make code harder to follow and can result in subtle bugs.

**Review by vasinov — CHANGES_REQUESTED** [review]

File: griptape/tools/date_time/tool.py [hunk 2]
```diff
@@ -34,6 +35,9 @@ def get_relative_datetime(self, params: dict) -> BaseArtifact:
             date_string = params["values"]["relative_date_string"]
             relative_datetime = parse(date_string)
 
-            return TextArtifact(str(relative_datetime))
+            if not relative_datetime:
```
Comment (vasinov) [inline griptape/tools/date_time/tool.py:38]: Sorry to nit more but let's also make sure `if-else` structures read "naturally" with a positive `if` coming first and a negative being naturally tucked under `else`. Makes it a little easier to follow the code when developers don't have to negate a condition in their head while reading the code.

**Review by vasinov — CHANGES_REQUESTED** [review]

File: griptape/tools/date_time/tool.py (hunk 2 above)
Comment (collindutter) [reply griptape/tools/date_time/tool.py]: Makes sense, updated!

**Review by vasinov — APPROVED** [review]

**Pull request merged**